sweepbatcher/sweep_batch.go

```diff
@@ -561,46 +520,151 @@ func (b *batch) addSweep(ctx context.Context, sweep *sweep) (bool, error) {
+	if numExisting == len(sweeps) {
+		for _, s := range sweeps {
+			oldSweep, ok := b.sweeps[s.outpoint]
+			if !ok {
+				return false, fmt.Errorf("sweep %v not found "+
+					"in batch %d", s.outpoint, b.id)
+			}
+
+			// Preserve coopFailed value not to forget about
+			// cooperative spending failure in this sweep.
+			tmp := *s
+			tmp.coopFailed = oldSweep.coopFailed
+
+			// If the sweep was resumed from storage, and the swap
+			// requested to sweep again, a new sweep notifier will
+			// be created by the swap. By re-assigning to the
+			// batch's sweep we make sure that everything, including
+			// the notifier, is up to date.
+			b.sweeps[s.outpoint] = tmp
+
+			// If this is the primary sweep, we also need to update
+			// the batch's confirmation target and fee rate.
+			if b.primarySweepID == s.outpoint {
+				b.cfg.batchConfTarget = s.confTarget
+				b.rbfCache.SkipNextBump = true
+			}
+
+			// Update batch's fee rate to be greater than or equal
+			// to minFeeRate of the sweep. Make sure batch's fee
+			// rate does not decrease (otherwise it won't pass RBF
+			// rules and won't be broadcasted) and that it is not
+			// lower that minFeeRate of other sweeps (so it is
+			// applied).
+			if b.rbfCache.FeeRate < s.minFeeRate {
+				b.rbfCache.FeeRate = s.minFeeRate
+			}
 		}
+
+		return true, nil
 	}
 
-	// Add the sweep to the batch's sweeps.
-	b.Infof("adding sweep %x", sweep.swapHash[:6])
-	b.sweeps[sweep.outpoint] = *sweep
+	// Here is the code to add new sweeps to a batch.
+	for _, s := range sweeps {
+		// If this is the first sweep being added to the batch, make it
+		// the primary sweep.
+		if b.primarySweepID == zeroSweepID {
+			b.primarySweepID = s.outpoint
+			b.cfg.batchConfTarget = s.confTarget
+			b.rbfCache.FeeRate = s.minFeeRate
+			b.rbfCache.SkipNextBump = true
+
+			// We also need to start the spend monitor for this new
+			// primary sweep.
+			err := b.monitorSpend(ctx, *s)
+			if err != nil {
+				return false, err
+			}
+		}
+
+		// Make sure the sweep is not present in the batch. If it is
+		// present, this is a bug, return an error to stop here.
+		if _, has := b.sweeps[s.outpoint]; has {
+			return false, fmt.Errorf("sweep %v is already present "+
+				"in batch %d", s.outpoint, b.id)
+		}
+
+		// Add the sweep to the batch's sweeps.
+		b.Infof("adding sweep %v, swap %x", s.outpoint, s.swapHash[:6])
+		b.sweeps[s.outpoint] = *s
 
-	// Update FeeRate. Max(sweep.minFeeRate) for all the sweeps of
-	// the batch is the basis for fee bumps.
-	if b.rbfCache.FeeRate < sweep.minFeeRate {
-		b.rbfCache.FeeRate = sweep.minFeeRate
-		b.rbfCache.SkipNextBump = true
+		// Update FeeRate. Max(s.minFeeRate) for all the sweeps of
+		// the batch is the basis for fee bumps.
+		if b.rbfCache.FeeRate < s.minFeeRate {
+			b.rbfCache.FeeRate = s.minFeeRate
+			b.rbfCache.SkipNextBump = true
+		}
+
+		if err := b.persistSweep(ctx, *s, false); err != nil {
+			return true, err
```

Comment:
does this count as accepted? Should we return `false` here?

Reply:
It was accepted in-memory, but failed to be persisted. Loop will shutdown with this error anyway.

Also this matches to the previous behavior:
```go
        return true, b.persistSweep(ctx, *sweep, false)
```